nucliadb_core/src/metrics/metric/tokio_runtime.rs

```diff
@@ -19,39 +19,55 @@
 
 use std::sync::{Mutex, MutexGuard, TryLockError};
 
+use crate::tracing::info;
 use prometheus_client::metrics::counter::Counter;
 use prometheus_client::metrics::gauge::Gauge;
 use prometheus_client::metrics::histogram::Histogram;
 use prometheus_client::registry::Registry;
+use tokio;
 use tokio::runtime::Handle;
 use tokio_metrics::{RuntimeIntervals, RuntimeMonitor};
 
 use crate::{node_error, NodeResult};
 
 pub struct TokioRuntimeObserver {
-    runtime: Handle,
-    intervals: Mutex<RuntimeIntervals>,
+    runtime: Option<Handle>,
+    intervals: Option<Mutex<RuntimeIntervals>>,
     metrics: TokioRuntimeMetrics,
 }
 
 impl TokioRuntimeObserver {
     pub fn new(registry: &mut Registry) -> Self {
-        let runtime = Handle::current();
-        let monitor = RuntimeMonitor::new(&runtime);
-        // We need to store RuntimeIntervals iterator instead of RuntimeMonitor to
-        // get incremental values. We need a Mutex to be Send and Sync (for our
-        // Meter users)
-        let intervals = Mutex::new(monitor.intervals());
-        Self {
-            runtime,
-            intervals,
-            metrics: TokioRuntimeMetrics::new(registry),
+        let runtime = Handle::try_current();
+        if runtime.is_err() {
+            info!("Cannot export tokio runtime metrics, no runtime available");
+            Self {
+                runtime: None,
+                intervals: None,
+                metrics: TokioRuntimeMetrics::new(registry),
+            }
+        } else {
+            let runtime = runtime.unwrap();
+            let monitor = RuntimeMonitor::new(&runtime);
+            // We need to store RuntimeIntervals iterator instead of RuntimeMonitor to
+            // get incremental values. We need a Mutex to be Send and Sync (for our
+            // Meter users)
+            let intervals = Mutex::new(monitor.intervals());
+            Self {
+                runtime: Some(runtime),
+                intervals: Some(intervals),
+                metrics: TokioRuntimeMetrics::new(registry),
+            }
         }
     }
 
     pub fn observe(&self) -> NodeResult<()> {
+        if self.runtime.is_none() {
+            return Ok(());
+        }
+
         let interval = self.next_interval()?;
-        let raw_metrics = self.runtime.metrics();
+        let raw_metrics = self.runtime.as_ref().unwrap().metrics();
 
         self.metrics.update(raw_metrics, interval);
```

Comment:
I would try to avoid this pattern of checking if something is the error variant or none and then unwrapping  for pattern maching checks, which are less error prone. For example here you could:
```rust
        let Some(runtime) = self.runtime.as_ref() else {
            return Ok(());
        };
        let interval = self.next_interval()?;
        let raw_metrics = runtime.metrics();
        self.metrics.update(raw_metrics, interval);

        Ok(())
```